pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -573,29 +573,47 @@ func (r *ReconcileLogStorage) elasticsearchSecrets(ctx context.Context) ([]*core
 	svcDNSNames := dns.GetServiceDNSNames(render.ElasticsearchServiceName, render.ElasticsearchNamespace, r.clusterDomain)
 
 	// Get the secret - might be nil
-	secret, err := utils.GetSecret(ctx, r.client, render.TigeraElasticsearchCertSecret, render.OperatorNamespace())
+	secret, err := render.GetSecret(ctx, r.client, render.TigeraElasticsearchCertSecret, render.OperatorNamespace())
 	if err != nil {
 		return nil, err
 	}
 
-	// Ensure that secret is valid.
-	secret, err = utils.EnsureCertificateSecret(ctx, render.TigeraElasticsearchCertSecret, secret, svcDNSNames...)
+	// Ensure that cert is valid.
+	secret, err = render.EnsureCertificateSecret(ctx, render.TigeraElasticsearchCertSecret, secret, "tls.key", "tls.crt", render.DefaultCertificateDuration, svcDNSNames...)
 	if err != nil {
 		return nil, err
 	}
 
 	secrets = append(secrets, secret, render.CopySecrets(render.ElasticsearchNamespace, secret)[0])
 
 	// Get the pub secret - might be nil
-	pubSecret, err := utils.GetSecret(ctx, r.client, render.ElasticsearchPublicCertSecret, render.ElasticsearchNamespace)
+	pubSecret, err := render.GetSecret(ctx, r.client, render.ElasticsearchPublicCertSecret, render.ElasticsearchNamespace)
 	if err != nil {
 		return nil, err
 	}
 
 	if pubSecret != nil {
-		secrets = append(secrets, render.CopySecrets(render.OperatorNamespace(), pubSecret)...)
+		var ok bool
+		ok, err = render.SecretHasExpectedDNSNames(pubSecret, "tls.crt", svcDNSNames)
+		if err != nil {
+			return nil, err
+		}
+		// If the cert was updated and the public cert already exists, delete it so
+		// it is recreated by eck operator.
+		if !ok {
+			obj := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: render.ElasticsearchPublicCertSecret, Namespace: render.ElasticsearchNamespace},
```

Comment:
Isnt' this just the pubSecret? Couldn't you just do r.client.Delete(ctx, pubSecret) insetead?

Reply:
Yes!